src/controllers/legacycontrollermapping.h

```diff
@@ -35,20 +42,34 @@ class LegacyControllerMapping {
               m_settingsLayout(other.m_settingsLayout.get() != nullptr
                               ? other.m_settingsLayout->clone()
                               : nullptr),
+#ifdef MIXXX_USE_QML
+              m_scripts(other.m_scripts),
+              m_modules(other.m_modules),
+              m_screens(other.m_screens) {
+#else
               m_scripts(other.m_scripts) {
+#endif
     }
     virtual ~LegacyControllerMapping() = default;
 
     virtual std::shared_ptr<LegacyControllerMapping> clone() const = 0;
 
     struct ScriptFileInfo {
+        enum Type {
+            JAVASCRIPT,
+#ifdef MIXXX_USE_QML
+            QML,
+#endif
+        };
```

Comment:
I did make some test already, and this will lead to failing loading the mapping due to an unsupported file type (which is the expected behaviour) . I kinda like the idea not to have any QML related symbol used or referenced in a QML-free build.

Reply:
Fair, I'm just saying that if there is other code that just does a `static_cast for example, it could end badly, but if the ifdefs are placed properly (which is not always the case unfortunately) it shouldn't be a problem.